builq.go

```diff
@@ -78,28 +61,40 @@ func (b *Builder) writeArgs(s fmt.State, verb rune, arg any, isMulti bool) {
 			fmt.Fprint(s, ", ")
 		}
 
-		b.appendArg(arg, verb)
-
 		switch verb {
 		case '$': // PostgreSQL
+			b.counter++
 			fmt.Fprintf(s, "$%d", b.counter)
 		case '?': // MySQL/SQLite
 			fmt.Fprint(s, "?")
 		default:
 			panic("unreachable")
 		}
+
+		// store the first placeholder used in the query
+		// to check for mixed placeholders later.
+		if b.placeholder == 0 {
+			b.placeholder = verb
+		}
+		if b.placeholder != verb {
+			b.err = errMixedPlaceholders
+			return
+		}
+
+		b.args = append(b.args, arg)
 	}
 }
 
-func (b *Builder) appendArg(arg any, placeholder rune) {
```

Comment:
Merged to `writeArgs` 'cause actually the only line related to appending args is the last one, everything else is about placeholders